RiotSwiftUI/Modules/Authentication/Common/Service/MatrixSDK/AuthenticationService.swift

```diff
@@ -148,8 +148,13 @@ class AuthenticationService: NSObject {
                 self.registrationWizard = registrationWizard
             } catch {
                 guard homeserver.preferredLoginMode.hasSSO, error as? RegistrationError == .registrationDisabled else {
-                    throw error
+                    if homeserver.preferredLoginMode.providesDelegatedOIDCCompatibility {
+                        throw RegistrationError.delegatedOIDCRequiresReplacementApp
```

Comment:
I'm mixed about this now I have implemented it. Maybe this shouldn't throw an error at all, instead letting the 2 screens read that `providesDelegatedOIDCCompatibility` property from the homeserver to show the banners. WDYT?

Reply:
It probably would've been nice but the current solution isn't that bad either so probably not worth the trouble to change it.